packages/meta/src/validatordef/parser/ValidatorCreators.ts

```diff
@@ -142,25 +137,26 @@ export function createExpressionRule(data: Partial<ExpressionRule>): ExpressionR
     return result;
 }
 
-export function createIsuniqueRule(data: Partial<IsUniqueRule>): IsUniqueRule {
+export function createIsUniqueRule(data: Partial<IsUniqueRule>): IsUniqueRule {
     const result = new IsUniqueRule();
 
     createRuleCommonParts(data, result);
-    if (!!data.list) {
-        result.list = data.list;
-    }
-    if (!!data.listproperty) {
-        // the expression parser/creator returns a FreLangConceptExp
-        // but we need a FreLangSelfExp where 'self' refers to an element of result.list
-        // therefore we change the received 'listproperty'.
-        result.listproperty = new FreLangSelfExp();
-        // result.listproperty.sourceName = "list";
-        result.listproperty.location = data.listproperty.location;
-        result.listproperty.appliedfeature = new FreLangAppliedFeatureExp();
-        result.listproperty.appliedfeature.sourceName = data.listproperty.sourceName;
-        result.listproperty.appliedfeature.location = data.listproperty.location;
-        result.listproperty.appliedfeature.appliedfeature = data.listproperty.appliedfeature;
-        result.listproperty.appliedfeature.sourceExp = result.listproperty;
+    if (!!data.listExp) {
+        result.listExp = data.listExp;
+    }
+    if (!!data.listpropertyExp) {
+        // // Normally, an expression would refer to a property of the surrounding classifier,
+        // // as in 'self.something' or 'something',
+        // // but here we need a property of type of the elements in the indicated list.
+        // // Therefore, we insert an extra FreVarExp in front of the property expression,
+        // // which refers to the correct classifier. The latter is determined in the checker.
+        // // So the 'self.something' or 'something' becomes 'list.self.something' or 'list.something'.
```

Comment:
Not quite sure why this is neccesary, but `list.self` looks strange.

- Is this just internal?  Or does the language engineer see this anywhere?
- Can the language engineer write this directly?